src/ucp/api/ucp.h

```diff
@@ -1089,8 +1089,10 @@ void ucp_worker_release_address(ucp_worker_h worker, ucp_address_t *address);
  * communication progress.
  *
  * @param [in]  worker    Worker to progress.
+ *
+ * @return Non-zero if any communication was progressed, zero otherwise.
```

Comment:
Using standard error codes would add more checks and branches, but when returning progress count can use simple add operation when calling multiple progress callbacks. this is how it's done in openmpi as well.